src/dftbp/dftb/periodic.F90

```diff
@@ -64,28 +64,34 @@ module dftbp_dftb_periodic
     !> whether the neighbour list has been set by an API call
     logical :: setExternally = .false.
 
-    ! Whether memory should be allocated via MPI-windows (or directly via allocate() otherwise)
+    !> Whether memory should be allocated via MPI-windows (or directly via allocate() otherwise)
+    !!
+    !! Note: this variable cannot be inside the MPI block below with current code, as the parser
+    !! uses this data structure before the MPI environment is set up (in addition to the main code
+    !! use). This leads to a fall back to the usual distributed case if MPI is enabled, instead of
+    !! the shared window in that case.
     logical, private :: useMpiWindows_ = .false.
 
-    ! memory allocated for the iNeighbour array
+  #:if WITH_MPI
```

Comment:
See my other comment. I think, we need the `*Mem_` also if MPI windows is not used.